src/hooks/useArticle.js

```diff
@@ -1,28 +1,27 @@
 import { useState, useEffect } from 'preact/hooks'
-import { useI18n } from 'hooks'
-import { getArticle, getArticleMedia, getSuggestedArticles } from 'api'
+import { getArticle, getArticleMedia, getSuggestedArticles, loadMessages } from 'api'
 import { canonicalizeTitle } from 'utils'
 
 export const useArticle = (lang, title) => {
   const [article, setArticle] = useState(null)
-  const i18n = useI18n()
 
   const loadArticle = () => {
     setArticle(null)
-    Promise.all([getArticle(lang, title), getArticleMedia(lang, title), getSuggestedArticles(lang, title)])
-      .then(([article, media, suggestedArticles]) => {
+    Promise.all([getArticle(lang, title), getArticleMedia(lang, title), getSuggestedArticles(lang, title), loadMessages(lang)])
+      .then(([article, media, suggestedArticles, messages]) => {
         const { sections, toc } = article
-        const footerTitle = i18n.i18n('toc-footer')
+        const footerTitle = messages[lang] && messages[lang]['suggested-articles'] ? messages[lang]['suggested-articles'] : messages.en['suggested-articles']

```

Comment:
We shouldn't do raw access to the i18n files data. That's what [banana-i18n](https://github.com/wikimedia/banana-i18n) is for. It has a bunch of features including proper fallback chains. Not everything falls back to english immediately. 

I would suggest changing the way we load messages to just load them all on startup (assuming it doesn't introduce a performance hit) and changing the locale of the instance temporarily to resolve the message. Note that we would have to be careful not to leave the instance with the wrong locale after this point.

```
i18n.setLocale(article.contentLang)
const footerTitle = i18n.i18n('toc-footer')
// restore i18n locale
```

Alternatively, we could request a feature in banana-i18n to specify the locale for a specific `i18n()` call. That's a very common scenario in MediaWiki where we always have to think if a specific message is best displayed in user or content language.

Reply:
I'm thinking of is that possible we maintain two i18n object, one for app language, one for article language?

Reply:
You would have to create a new one every time you change the app or article language. I would suggest having only one instance with all messages in it. We already need to do it for the fallback chains to work correctly.

We have a related ticket to load messages faster: https://phabricator.wikimedia.org/T236898

Reply:
by doing that assuming we have one object contains all the messages `{ en: {}, de: {}, hi: {}, fr: {} ... }` we are able to load all `banana.load( <all messages>, <locale> )`

I don't see a way to load two different locales using single banana `banana.i18n()`, it doesn't
allow settings locale for each call of `i18n()`, `banana.setLocal()` can be used but we don't want to keep switching before we use for content or app language.

are there some other features I have missed in banana?

> Alternatively, we could request a feature in banana-i18n to specify the locale for a specific i18n() call. That's a very common scenario in MediaWiki where we always have to think if a specific message is best displayed in user or content language.

+1 for this as this will be very useful

Reply:
Great feedback all around, thank you guys. Just pushed a couple of commits that follow Stephane's suggestion. Few thoughts: 

* Using `setLocale` seems indeed cleaner inside `useArticle`
* Fallback chains seems to be working properly, for example:
    * bcl.json exists but doesn’t have `toc-footer` key
        * if I change cat article language to Central Bicol/Bikol, then label falls back to default  “Suggested articles” as expected
    * af.json does not exist
        * if I change cat article language to Afrikaans, then label falls back to “Suggested articles” as expected
* But am I missing something else regarding fallback chains?
* The performance seems to be _slightly_ hit on initial load but not in a deterrent way. What do you think? We can find a way to measure precisely if we want to be accurate